src/highdicom/legacy/sop.py

```diff
@@ -23,533 +24,385 @@
-            )
-
-        sop_class_uid = LEGACY_ENHANCED_SOP_CLASS_UID_MAP[ref_ds.SOPClassUID]
-
-        super().__init__(
-            study_instance_uid=ref_ds.StudyInstanceUID,
-            series_instance_uid=series_instance_uid,
-            series_number=series_number,
-            sop_instance_uid=sop_instance_uid,
-            sop_class_uid=sop_class_uid,
-            instance_number=instance_number,
-            manufacturer=ref_ds.Manufacturer,
-            modality=ref_ds.Modality,
-            transfer_syntax_uid=None,  # FIXME: frame encoding
-            patient_id=ref_ds.PatientID,
-            patient_name=ref_ds.PatientName,
-            patient_birth_date=ref_ds.PatientBirthDate,
-            patient_sex=ref_ds.PatientSex,
-            accession_number=ref_ds.AccessionNumber,
-            study_id=ref_ds.StudyID,
-            study_date=ref_ds.StudyDate,
-            study_time=ref_ds.StudyTime,
-            referring_physician_name=ref_ds.ReferringPhysicianName,
-            **kwargs
-        )
-        _convert_legacy_to_enhanced(legacy_datasets, self)
-
-
-class LegacyConvertedEnhancedPETImage(SOPClass):
-
+                distinguishing_tags_missing.append(tg)
+        logger_msg = set()
+        for ds in self.MixedFramesCopy:
+            all_equal = True
+            for tg in distinguishing_tags_missing:
+                if tg in ds:
+                    logger_msg.add(
+                        '{} is missing in all but {}'.format(
+                            DicomHelper.tag2kwstr(tg), ds['SOPInstanceUID']))
+                    all_equal = False
+                    break
+            if not all_equal:
+                continue
+            for tg in distinguishing_tags_existing:
+                ref_val = similar_ds[0][tg].value
+                if tg not in ds:
+                    all_equal = False
+                    break
+                new_val = ds[tg].value
+                if not DicomHelper.isequal(ref_val, new_val):
+                    logger_msg.add(
+                        'Inequality on distinguishing '
+                        'attribute{} -> {} != {} \n series uid = {}'.format(
+                            DicomHelper.tag2kwstr(tg), ref_val, new_val,
+                            ds.SeriesInstanceUID))
+                    all_equal = False
+                    break
+            if all_equal:
+                similar_ds.append(ds)
+        for msg_ in logger_msg:
+            logger.info(msg_)
+        for ds in similar_ds:
+            if ds in self.MixedFramesCopy:
+                self.MixedFramesCopy.remove(ds)
+        return (similar_ds, distinguishing_tags_existing)
+
+    @property
+    def DistinguishingAttributeKeywords(self) -> List[str]:
+        return self._DistinguishingAttributeKeywords[:]
+
+    @property
+    def FrameSets(self) -> List[FrameSet]:
+        return self._FrameSets
+
+
+class LegacyConvertedEnhanceImage(SOPClass):
```

Comment:
Classes derived from `SOPClass` should actually map to a DICOM SOP Class (and corresponding IOD). There is no DICOM *Legacy Converted Enhance Image* IOD!

This module previously defined three classes: `LegacyConvertedEnhancedMRImage`, `LegacyConvertedEnhancedCTImage` and `LegacyConvertedEnhancedPETImage`. We need those three separate classes.